pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -368,7 +368,9 @@ func (c *Controller) injectPacket(tf *opsv1alpha1.Traceflow) error {
 	if tf.Spec.Packet.TransportHeader.TCP != nil {
 		srcTCPPort = uint16(tf.Spec.Packet.TransportHeader.TCP.SrcPort)
 		dstTCPPort = uint16(tf.Spec.Packet.TransportHeader.TCP.DstPort)
-		flagsTCP = uint8(tf.Spec.Packet.TransportHeader.TCP.Flags)
+		if flagsTCP == 0 {
+			flagsTCP = uint8(tf.Spec.Packet.TransportHeader.TCP.Flags)
```

Comment:
I didn't give it much thought but should you use an "OR" operation here instead of ignoring user-provided flags if `flagsTCP != 0` (I presume this is in the case of the destination being a Service)? @tnqn

Reply:
Changed here to below to avoid ignoring user-provided flags:
```
		if tf.Spec.Packet.TransportHeader.TCP.Flags != 0 {
			flagsTCP = uint8(tf.Spec.Packet.TransportHeader.TCP.Flags)
		}
```